src/components/AddBook.js

```diff
@@ -1,12 +1,42 @@
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { add } from '../redux/books/books';
 
-const AddBook = () => (
-  <form>
-    <h2>Add a book</h2>
-    <input type="text" placeholder="Title" name="title" />
-    <input type="text" placeholder="Author" name="author" />
-    <button type="submit">Add</button>
-  </form>
-);
+const AddBook = () => {
+  const bookList = useSelector((state) => state);
+  const dispatch = useDispatch();
 
+  const submit = (e) => {
+    e.preventDefault();
+    const title = e.target.title.value;
+    const author = e.target.author.value;
```

Comment:
Kindly, define local state out side the `submit` method like

```
const [title, setTitle] = useState('');
```
and set title and author on every input update (onChange) rather than setting the state in submit.